### PR TITLE
SDK-56: Java SDK support for Spark Job Server

### PR DESCRIPTION
**Pull request by abhisheksr8**
Support for spark job server

### CODE REVIEW

**Comment by amoghmargoor** [comment]
Can we rename App to SparkJobServerApp everywhere ?

**Comment by amoghmargoor** [comment]
Also api and entities package are just bloating... Can we start organising them with this PR ....We will introduce new folders under api and entities to hold their respective stuff ...We need to clean up old things too someday, but thats a major compatibility change ... So we can club that with some genuine compatibility change that we push out in future.

File: src/main/java/com/qubole/qds/sdk/java/details/InvocationCallbackBase.java [hunk 1]
```diff
@@ -62,7 +62,6 @@ public final RawInvokableBuilder raw()
         {
             return invokeArguments.getClient().invokeRequest(invokeArguments.getForPage(), invokeArguments.getEntity(), invokeArguments.getGenericResponseType(), invokeArguments.getAdditionalPaths());
         }
-
```
Comment (amoghmargoor) [inline src/main/java/com/qubole/qds/sdk/java/details/InvocationCallbackBase.java]: Remove changes from this file.

Comment (abhisheksr8) [inline src/main/java/com/qubole/qds/sdk/java/details/InvocationCallbackBase.java]: done

File: src/main/java/com/qubole/qds/sdk/java/details/QdsClientImpl.java [hunk 2]
```diff
@@ -56,10 +57,11 @@
     private final DbTapApiImpl dbTapsApi;
     private final ReportApiImpl reportApi;
     private final SchedulerApiImpl schedulerApi;
+    private final AppApiImpl appApi;
 
     private static final ObjectMapper MAPPER = new ObjectMapper();
 
-    static ObjectMapper getMapper()
+    public static ObjectMapper getMapper()
```
Comment (abhisheksr8) [inline src/main/java/com/qubole/qds/sdk/java/details/QdsClientImpl.java]: this is used across package in AppBuilderImpl, also will be needed in future once we start moving specific things in specific folder

Comment (amoghmargoor) [inline src/main/java/com/qubole/qds/sdk/java/details/QdsClientImpl.java]: Ok actually my question was why is even being shared as there can be threading issues. But looks like ObjectMapper is safe and can be shared across different threads., so thats fine.

Comment (abhisheksr8) [inline src/main/java/com/qubole/qds/sdk/java/details/QdsClientImpl.java]: ok, yes it is thread safe
http://wiki.fasterxml.com/JacksonFAQThreadSafety

File: src/main/java/com/qubole/qds/sdk/java/details/GenericInvokableBuilderImpl.java [hunk 3]
```diff
@@ -19,7 +19,7 @@
 import com.qubole.qds.sdk.java.client.QdsClient;
 import javax.ws.rs.core.GenericType;
 
-class GenericInvokableBuilderImpl<T> extends InvocationCallbackBase<T> implements InvokableBuilder<T>
```
Comment (amoghmargoor) [inline src/main/java/com/qubole/qds/sdk/java/details/GenericInvokableBuilderImpl.java]: This becomes a huge problem to make it public. Making it public here means we have to take care of it's compatibility too with version on version like we do to public API's and document it too etc etc .. It now becomes a public API to be used by users. I understand it's required here but any thoughts upon the caveats ?

Comment (abhisheksr8) [inline src/main/java/com/qubole/qds/sdk/java/details/GenericInvokableBuilderImpl.java]: yes there is no subpackage level visibility in java, so i think we should avoid the subfoldering, as there is no way to access non public classes in separate folder.

also generally users entry point is via QdsClientImpl object, ideally no one should try to access GenericInvokableBuilderImpl directly, but still better we shouldnt expose this to them.
so should i revert the sub packaging changes until we figure out if or how we should go about it?

Comment (amoghmargoor) [inline src/main/java/com/qubole/qds/sdk/java/details/GenericInvokableBuilderImpl.java]: yeah ...it was not designed properly earlier ... just revert changes for details folder

Comment (abhisheksr8) [inline src/main/java/com/qubole/qds/sdk/java/details/GenericInvokableBuilderImpl.java]: thanks, done

**Comment by amoghmargoor** [comment]
@abhisheksr8 Travis build failed. Please check.

**Comment by abhisheksr8** [comment]
fixed build

**Pull request merged**